src/Compilers/VisualBasic/Portable/SymbolDisplay/SymbolDisplayVisitor.vb

```diff
@@ -12,51 +12,77 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
     Partial Friend Class SymbolDisplayVisitor
         Inherits AbstractSymbolDisplayVisitor
 
-        Private ReadOnly _escapeKeywordIdentifiers As Boolean
+        Private Shared ReadOnly s_visitorPool As New ObjectPool(Of SymbolDisplayVisitor)(Function(pool) New SymbolDisplayVisitor(pool), 128)
+
+        Private ReadOnly _pool As ObjectPool(Of SymbolDisplayVisitor)
+        Private _escapeKeywordIdentifiers As Boolean
 
         ' A Symbol in VB might be a PENamedSymbolWithEmittedNamespaceName and in that case the 
         ' casing of the contained types and namespaces might differ because of merged classes/namespaces.
         ' To maintain the original spelling an emittedNamespaceName with the correct spelling is passed to 
         ' this visitor. 
 
-        Friend Sub New(
+        Private Sub New(pool As ObjectPool(Of SymbolDisplayVisitor))
+            _pool = pool
+        End Sub
+
+        Friend Shared Function GetInstance(
             builder As ArrayBuilder(Of SymbolDisplayPart),
             format As SymbolDisplayFormat,
             semanticModelOpt As SemanticModel,
-            positionOpt As Integer)
-
-            MyBase.New(builder, format, True, semanticModelOpt, positionOpt)
-            Debug.Assert(format IsNot Nothing, "Format must not be null")
+            positionOpt As Integer) As SymbolDisplayVisitor
 
-            Me._escapeKeywordIdentifiers = format.MiscellaneousOptions.IncludesOption(SymbolDisplayMiscellaneousOptions.EscapeKeywordIdentifiers)
-        End Sub
+            Dim instance = s_visitorPool.Allocate()
+            instance.Initialize(builder, format, isFirstSymbolVisited:=True, semanticModelOpt, positionOpt, inNamespaceOrType:=False)
+            Return instance
+        End Function
 
-        Private Sub New(
+        Private Shared Function GetInstance(
             builder As ArrayBuilder(Of SymbolDisplayPart),
             format As SymbolDisplayFormat,
             semanticModelOpt As SemanticModel,
             positionOpt As Integer,
             escapeKeywordIdentifiers As Boolean,
             isFirstSymbolVisited As Boolean,
-            Optional inNamespaceOrType As Boolean = False)
+            inNamespaceOrType As Boolean) As SymbolDisplayVisitor
 
-            MyBase.New(builder, format, isFirstSymbolVisited, semanticModelOpt, positionOpt, inNamespaceOrType)
+            Dim instance = s_visitorPool.Allocate()
+            instance.Initialize(builder, format, isFirstSymbolVisited, semanticModelOpt, positionOpt, inNamespaceOrType)
+            instance._escapeKeywordIdentifiers = escapeKeywordIdentifiers
+            Return instance
+        End Function
 
-            Me._escapeKeywordIdentifiers = escapeKeywordIdentifiers
+        Protected Overloads Sub Initialize(builder As ArrayBuilder(Of SymbolDisplayPart), format As SymbolDisplayFormat, isFirstSymbolVisited As Boolean, semanticModelOpt As SemanticModel, positionOpt As Integer, inNamespaceOrType As Boolean)
```

Comment:
>Overloads [](http://example.com/codeflow?start=18&length=9)

```New```?

Reply:
I have no idea what the difference is here. When I removed `Overrides`, there was a code fix that suggested I add `Overloads` so that's what I did.

Isn't `New` a constructor in VB?

Reply:
I guess I meant to suggest `Shadows`

Reply:
➡️ Now uses `Shadows`, and filed #70434 to improve the code fix